server-spi/src/main/java/org/keycloak/utils/KeycloakSessionUtil.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.utils;
+
+import org.keycloak.common.util.Resteasy;
+import org.keycloak.models.KeycloakSession;
+
+public class KeycloakSessionUtil {
```

Comment:
Perhaps the next round of refactorings is to remove this class and the possibility of obtaining the session by means other than when creating providers. Or at least not expose this (or mark as not supported) to extensions.

Reply:
It's certainly possible to remove it, and force our code and users to pass the current session everywhere that it's needed. That seemed like a much bigger change to take on - will you capture an issue for it?

Reply:
https://github.com/keycloak/keycloak/issues/29107